recovery/recovery.qdoc

```diff
@@ -203,7 +207,19 @@
   -# runs custom recovery (or built-in implementation of it) against merged results.
 
 
-  Built-in implementation of custom recovery for each key from merged results:\n
+  Built-in implementation of custom recovery for each key from merged results consists of two steps: server-send recovery with following rest-keys recovery.\n
+  If key could not be recovered via server-send, then it will be processed during rest-keys recovery.\n
+
+  Server-send recovery:
+  -# while there are non-empty buckets:
+    -# get next bucket - file that contains newest keys from a specific group
+    -# iterate over bucket keys:
+      -# check that key could be recovered via server-send, if so:
+        -# get destination groups to which key must be recovered and call server-send for it
+        -# check server-send result for the key: if operation failed, then move key to the other bucket if possible, otherwise to 'rest-keys'
```

Comment:
Can you describe how a key migrates between different buckets including 'rest_keys' bucket.